### PR TITLE
DBZ-357 Support for test execution against MariaDB

### PR DESCRIPTION
**Pull request by jpechane**

### CODE REVIEW

File: debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlConnectorConfig.java [hunk 1]
```diff
@@ -406,15 +454,14 @@ public static SecureConnectionMode parse(String value, String defaultValue) {
...
-    public static final Field JDBC_DRIVER = Field.create("database.jdbc.driver")
-                                                .withDisplayName("Jdbc Driver Class Name")
-                                                .withType(Type.CLASS)
-                                                .withWidth(Width.MEDIUM)
-                                                .withDefault(com.mysql.jdbc.Driver.class.getName())
-                                                .withImportance(Importance.LOW)
-                                                .withValidation(Field::isClassName)
-                                                .withDescription("JDBC Driver class name used to connect to the MySQL database server.");
-    /**
+    public static final Field VENDOR = Field.create("mysql.vendor")
```
Comment (gunnarmorling) [inline debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlConnectorConfig.java]: What's the reasoning for introducing this over the existing "driver" property?

We could still use the new enum internally, parsed from the given driver class name. I'd prefer though to not break compatability with existing users of the driver option.

File: jenkins-jobs/mariadb-test.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
+# Job definition to build and deploy snapshot artifact
+---
+- job:
+    project-type: freestyle
+    name: debezium-mysql-test-mariadb
+    display-name: MariaDB Test
+    description: Executes tests fro MySQL Connector with MariaDB
```
Comment (gunnarmorling) [inline jenkins-jobs/mariadb-test.yml]: s/fro/for/

File: pom.xml [hunk 3]
```diff
@@ -245,6 +247,11 @@
                 <version>${version.mysql.driver}</version>
             </dependency>
             <dependency>
```
Comment (gunnarmorling) [inline pom.xml]: Will this also be added to the plug-in ZIP?

File: debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlConnectorConfig.java (hunk 1 above)
Comment (jpechane) [reply debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlConnectorConfig.java]: There could not be any users of driver property as it goes hand-in-hand with hard-coded JDBC url. So albeit this is superficially a configrable value it is effectivelly a constant. And as JDBC URL pattern and driver must be changed in accord I thought it is best to replace it with a single logical constant.

File: pom.xml (hunk 3 above)
Comment (jpechane) [reply pom.xml]: Yes, it will be added too - as it is a non-`provided` dependency.

**Comment by gunnarmorling** [comment]
Hey @jpechane, should we close this for the time being? It seems we won't be able to properly support MariaDB until either the binlog client has been updated upstream or we create a fork of it.

**Comment by jpechane** [comment]
I agree

**Pull request closed without merging**